modules/swagger-core/src/main/java/io/swagger/v3/core/jackson/ModelResolver.java

```diff
@@ -1519,6 +1519,11 @@ protected void applyBeanValidatorAnnotations(Schema property, Annotation[] annot
                 ArraySchema sp = (ArraySchema) property;
                 sp.setMinItems(size.min());
                 sp.setMaxItems(size.max());
+            }else if(openapi31 && property instanceof JsonSchema){
```

Comment:
This is not sufficient I believe, we need to basically replace the `instanceof` checks with a test on the type and types. It would probably make sense to have some helper method working for both 3.0 and 3.1

so something like

```
boolean isStringSchema(Schema schema) {
 return "string".equals(schema.getType()) ||" (schema.getTypes() != null && schema.getTypes().contains("string")
}
```

and so on.
you would then use that instead e.g. of `else if (property instanceof StringSchema) {`

the same goes for array, number etc.

Also it would be probably a series of `if`, not `else`, as in oas 3.1 we could have type = "integer, string" so both can be applied

minor: please add spaces after and before `{` and `}`

The same would apply below for pattern